modules/virtual_network_gateway/main.tf

```diff
@@ -97,22 +97,25 @@ resource "azurerm_virtual_network_gateway" "this" {
 # https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/public_ip
 resource "azurerm_public_ip" "this" {
   for_each = { for ip_configuration in var.ip_configuration :
-    ip_configuration.name => try(ip_configuration.public_ip_standard_sku, false)
+    ip_configuration.name => {
+      name                   = ip_configuration.public_ip_name
+      public_ip_standard_sku = ip_configuration.public_ip_standard_sku
+    }
   if ip_configuration.create_public_ip }
 
   resource_group_name = var.resource_group_name
   location            = var.location
-  name                = "${var.name_prefix}pip-vgw${var.name_suffix}-${each.key}"
+  name                = each.value.name
 
-  allocation_method = each.value ? "Static" : "Dynamic"
-  zones             = var.enable_zones ? var.avzones : null
+  allocation_method = each.value.public_ip_standard_sku ? "Static" : "Dynamic"
+  zones             = try(length(var.avzones) > 0, false) ? var.avzones : null
   tags              = var.tags
```

Comment:
place tags as the bottom of a resource - that's the convention in azure resource we have for a long time

Reply:
ok, fixed